azkaban-common/src/main/java/azkaban/Constants.java

```diff
@@ -110,6 +110,17 @@
     public static final String AZKABAN_KERBEROS_PRINCIPAL = "azkaban.kerberos.principal";
     public static final String AZKABAN_KEYTAB_PATH = "azkaban.keytab.path";
     public static final String PROJECT_TEMP_DIR = "project.temp.dir";
+
+    /*
+     * The max number of artifacts retained per project.
+     * Example: If this config is set to 3, then we have at latest 3 versions saved in storage.
+     * Constraints:
+     * - Has to be Integer
+     * - Negative values are not allowed and will crash the app.
+     * - A value of 0 implies all artifacts will be stored [Default]
```

Comment:
Can we have a value like None or False to imply all artifacts are stored?

Setting "azkaban.storage.artifact.max.retention=0" makes me think no artifacts are being retained.

Or maybe introduce a boolean true/false config corresponding to whether or not any artifacts are being cleaned?